src/Model.jl

```diff
@@ -80,13 +96,13 @@ function Model(lattice::AbstractMatrix{T};
                atoms=[],
                magnetic_moments=[],
                terms=[],
-               temperature=T(0.0),
+               temperature::Union{T, Unitful.Quantity}=T(0.0),
```

Comment:
I would not type-annotate this. If someone uses a very strange type, it'll just crash in the conversion.